### PR TITLE
fix(wallet): Added adaptor to properly refresh send modal total balances

### PR DESCRIPTION
**Pull request by Cuteivist**
Fixes https://github.com/status-im/status-desktop/issues/15306

### What does the PR do

* Added send modal adaptor component to properly expose asset models and balances
* Updated storybook with send modal adaptor page
* Updated storybook checkbox with optional exclusive button group

### Affected areas

Send modal

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/11396062/339a3eeb-8be9-49c2-8ae9-855ee7f17c0d

### CODE REVIEW

**Comment by caybro** [comment]
Potentially fixing issue https://github.com/status-im/status-desktop/issues/14768 too?

File: ui/imports/shared/popups/send/models/SendModalAssetsAdaptor.qml [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                model: filteredBalances
+                initialValue: "0"
+                roleName: "balance"
+
+                aggregateFunction: (aggr, value) => AmountsArithmetic.sum(
+                                       AmountsArithmetic.fromString(aggr),
+                                       AmountsArithmetic.fromString(value)).toString()
+            }
+
+            FunctionAggregator {
```
Comment (caybro) [inline ui/imports/shared/popups/send/models/SendModalAssetsAdaptor.qml:147]: What is this `chainsAggregator` used for?

File: ui/imports/shared/popups/send/models/SendModalAssetsAdaptor.qml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+             "icon", "visible", "marketDetailsAvailable", "marketDetailsLoading",
+             "marketPrice", "marketChangePct24hour", "isCommunityAsset", "balancesModel"]
+
+
+        /* Internal function to search token address */
+        function __searchAddressInList(addressPerChain, searchString) {
+            let addressFound = false
+            let tokenAddresses = ModelUtils.modelToFlatArray(addressPerChain, "address")
+            for (let i =0; i< tokenAddresses.length; i++){
+                if(tokenAddresses[i].toUpperCase().startsWith(searchString.toUpperCase())) {
```
Comment (caybro) [inline ui/imports/shared/popups/send/models/SendModalAssetsAdaptor.qml:174]: Could be optimized... why call `searchString.toUpperCase()` again and again for every `for` iteration?

**Review by caybro — APPROVED** [review]
LGTM, just 2 minor things

**Review by saledjenic — APPROVED** [review]

**Pull request merged**